### PR TITLE
[SNMP Traps] Add support for BITS field enrichment

### PR DESCRIPTION
**Pull request by ken-schneider**
What does this PR do?
Adds support for resolving bit enumerations to their string values assuming such a mapping exists in the traps database.

Motivation
Enriches data for our customers so they don't have to reference MIBs to determine what the value of a trap variable actually means.

Additional Notes
There will be another PR out for integrations-core which adds the bits enum definitions to the traps database.

Possible Drawbacks / Trade-offs
Initial start up may take longer as the default traps_db will be increasing in size. This also increases the memory footprint of the application. There are only ~230 variables with `bits`, thus size remains at 12MB uncompressed, and increases slightly to 1.5MB compressed.

There is a bit more processing that must be done to perform matching between the enum and the variable, but since we're using a map this should be negligible.

Describe how to test/QA your changes
Unit tests have been added to ensure the new traps_db format is accepted by the application and that bit enums are being properly handled.

To test end to end, you'll need an updated version of the traps_db which you can generate it yourself using the branch of integrations-core with my changes, or wait for that PR to be merged. Replace the existing traps_db on your agent with the updated one containing enums. Run the agent and send a trap like the following:

```sh
sudo snmptrap -v2c -c <community string> <ip-of-agent>:<port of agent (probably 9162)> '' 1.3.6.1.6.3.1.1.5.4 1.3.6.1.2.1.2.2.1.1 i 9001 1.3.6.1.2.1.2.2.1.7 i 2 1.3.6.1.2.1.2.2.1.8 i 7 1.3.6.1.4.1.2696.3.2.1.2.1.1.7 b 0,8,12,60,88,89
```
The above trap is a `LinkUp` trap with the following trap variables and values:
- ifAdminStatus: down
- ifIndex: 9001
- ifOperStatus: lowerLayerDown
- trapControlFailureSummary:
    - tsTsSyncLoss
    - tsPcrRepetitionError
    - tsCatError
    - tRFAccuracy
    - tSepSeti
    - 89 (this is not defined in the MIB, but it's sent and should show up in the UI)

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by jtappa — APPROVED** [review]
👍🏻 from docs team for release note

**Review by AlexandreYang — APPROVED** [review]

File: pkg/snmp/traps/traps_db.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ type VariableMetadata struct {
 	Name               string         `yaml:"name" json:"name"`
 	Description        string         `yaml:"descr" json:"descr"`
 	Enumeration        map[int]string `yaml:"enum" json:"enum"`
+	Bits               map[int]string `yaml:"bits" json:"bits"`
```
[the remark this answers is not in the document]
Comment (ken-schneider) [reply pkg/snmp/traps/traps_db.go:13]: It comes from the traps DB, the bits field is new as of this PR in `integrations-core` https://github.com/DataDog/integrations-core/pull/12581

The underlying logic is pretty much identical to parsing integer enums.

File: pkg/snmp/traps/formatter.go [hunk 2]
```diff
@@ -206,23 +196,56 @@ func IsValidOID(value string) bool {
...
+		log.Warnf("unable to enrich variable %q %s with BITS mapping, received value was not []byte, was %T", varMetadata.Name, variable.OID, variable.Value)
+		return variable.Value
+	}
+	enabledValues := make([]interface{}, 0)
+	for i, b := range bytes {
+		for j := 0; j < 8; j++ {
+			position := j + i*8 // position is the index in the current byte plus 8 * the position in the byte array
+			enabled, err := isBitEnabled(uint8(b), j)
+			if err != nil {
+				log.Debugf("unable to determine status at position %d: %s", position, err.Error())
```
[the remark this answers is not in the document]
Comment (ken-schneider) [reply pkg/snmp/traps/formatter.go:232]: Yeah, I agree it's a good idea to be explicit here.

File: pkg/snmp/traps/formatter.go [hunk 3]
```diff
@@ -206,23 +196,56 @@ func IsValidOID(value string) bool {
...
+	}
+
+	// if no mapping is found or type is not integer
+	log.Debugf("unable to find enum mapping for value %d variable %q", i, varMetadata.Name)
+	return variable.Value
+}
+
+// enrichBits checks to see if the variable has a mapping in bits and
+// returns the mapping if it exists, otherwise returns the value unchanged
+func enrichBits(variable trapVariable, varMetadata VariableMetadata) interface{} {
```
[the remark this answers is not in the document]
Comment (ken-schneider) [reply pkg/snmp/traps/formatter.go:219]: Added tests for it

**Pull request merged**